trl/trainer/dpo_trainer.py

```diff
@@ -374,42 +377,321 @@ def _prepare_deepspeed(self, model: PreTrainedModelWrapper):
         model.eval()
         return model
 
-    def concatenated_inputs(self, batch: Dict[str, Union[List, torch.LongTensor]]) -> Dict[str, torch.LongTensor]:
+    def get_train_dataloader(self) -> DataLoader:
+        # tokenize the dataset and compute reference logps for training datasets
+        self.train_dataset = self.train_dataset.map(self.tokenize_batch_element)
+        if self.precompute_ref_log_probs:
```

Comment:
Maybe it should be `if not self.precompute_ref_log_probs` here? Same in `get_eval_data_loader`

Reply:
opps so i wanted to keep the old default behaviour ... let me fix the eval data loader

Reply:
no its also fine in the `get_eval_dataloader`

Reply:
My mistake. Great jobs. Hope this could be merged soon.